src/Symfony/Bundle/FrameworkBundle/Command/KernelDebugCommand.php

```diff
@@ -0,0 +1,108 @@
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        /** @var $kernel KernelInterface */
+        $kernel = $this->getContainer()->get('kernel');
+        $baseDir = realpath($kernel->getRootDir().DIRECTORY_SEPARATOR.'..');
+        $bundles = array_map(function ($bundle) use ($baseDir) {
+            return '* '.$bundle->getName().' (<comment>'.$this->formatPath($bundle->getPath(), $baseDir).'</comment>)';
+        }, $kernel->getBundles());
+        sort($bundles);
+
+        $io->title('Welcome to Symfony '.Kernel::VERSION);
+        $io->success(array(
+            'Your application is now ready. You can start working on it at:',
+            $baseDir,
+        ));
+        $io->table(array(), array(
+            array('Kernel', get_class($kernel)),
+            array('Name', $kernel->getName()),
+            array('Version', Kernel::VERSION),
+            array('Version ID', Kernel::VERSION_ID),
+            array('End of maintenance', Kernel::END_OF_MAINTENANCE.($this->isExpired(Kernel::END_OF_MAINTENANCE) ? ' <error>Expired</error>' : '')),
+            array('End of life', Kernel::END_OF_LIFE.($this->isExpired(Kernel::END_OF_LIFE) ? ' <error>Expired</error>' : '')),
+            new TableSeparator(),
+            array('Environment', $kernel->getEnvironment()),
+            array('Debug', $kernel->isDebug() ? 'true' : 'false'),
+            array('Charset', $kernel->getCharset()),
+            array('Timezone', date_default_timezone_get().' (<comment>'.(new \DateTime())->format(\DateTime::W3C).'</comment>)'),
+            new TableSeparator(),
+            array('Root directory', $this->formatPath($kernel->getRootDir(), $baseDir)),
+            array('Cache directory', $this->formatPath($kernel->getCacheDir(), $baseDir).' (<comment>'.$this->formatFilesize($kernel->getCacheDir()).'</comment>)'),
+            array('Log directory', $this->formatPath($kernel->getLogDir(), $baseDir).' (<comment>'.$this->formatFilesize($kernel->getLogDir()).'</comment>)'),
+            new TableSeparator(),
+            array('Bundles (<comment>'.count($bundles).'</comment>)', implode(PHP_EOL, $bundles)),
+        ));
+    }
+
+    private function formatPath($path, $baseDir = null)
+    {
+        return null !== $baseDir ? preg_replace('~^'.preg_quote($baseDir, '~').'~', '.', $path) : $path;
+    }
+
+    private function formatFilesize($path)
+    {
+        if (is_file($path)) {
+            $size = filesize($path) ?: 0;
+        } else {
+            $size = 0;
+            foreach (new \RecursiveIteratorIterator(new \RecursiveDirectoryIterator($path, \RecursiveDirectoryIterator::SKIP_DOTS | \RecursiveDirectoryIterator::FOLLOW_SYMLINKS)) as $file) {
+                $size += $file->getSize();
+            }
+        }
+
+        static $units = array('B', 'kB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB');
```

Comment:
I'd prefer MB notation for filesize rather than MiB. `formatMemory($filesize)` feels weird :) We could move it entirely to the filesystem component, but for now i did it quick.